guide/src/conversions.md

```diff
@@ -1,8 +1,103 @@
+| `Iterator[Any]` | -                             | `&PyIterator`        |
+| `bytearray`   | -                               | `&PyByteArray`       |
+| `slice`       | -                               | `&PySlice`           |
+| `type`        | -                               | `&PyType`            |
+| `module`      | -                               | `&PyModule`          |
+| `datetime.datetime` | -                         | `&PyDateTime`        |
+| `datetime.date` | -                             | `&PyDate`            |
+| `datetime.time` | -                             | `&PyTime`            |
+| `datetime.tzinfo` | -                           | `&PyTzInfo`          |
+| `datetime.timedelta` | -                        | `&PyDelta`           |
+
+There are also a few special types related to the GIL and Rust-defined `#[pyclass]`es which may come in useful:
+
+| What          | Description |
+| ------------- | ------------------------------- |
+| `Python`      | A GIL token, used to pass to PyO3 constructors to prove ownership of the GIL |
+| `PyObject`    | A Python object isolated from the GIL lifetime. This can be sent to other threads. To call Python APIs using this object, it must be used with `AsPyRef::as_ref` to get a `&PyAny` reference. |
+| `Py<T>`       | Same as above, for a specific Python type or `#[pyclass]` T. |
+| `&PyCell<T>`  | A `#[pyclass]` value owned by Python. |
+| `PyRef<T>`    | A `#[pyclass]` borrowed immutably. |
+| `PyRefMut<T>` | A `#[pyclass]` borrowed mutably. |
+
+For more detail on accepting `#[pyclass]` values as function arguments, see [the section of this guide on Python Classes](class.md).
+
+#### Using Rust library types vs Python-native types
+
+Using Rust library types as function arguments will incur a conversion cost compared to using the Python-native types. Using the Python-native types is almost zero-cost (they just require a type check similar to the Python builtin function `isinstance()`).
+
+However, once that conversion cost has been paid, the Rust standard library types offer a number of benefits:
+- You can write functionality in native-speed Rust code (free of Python's runtime costs).
+- You get better interoperability with the rest of the Rust ecosystem.
+- You can use `Python::allow_threads` to release the Python GIL and let other Python threads make progress while your Rust code is executing.
+- You also benefit from stricter type checking. For example you can specify `Vec<i32>`, which will only accept a Python `list` containing integers. The Python-native equivalent, `&PyList`, would accept a Python `list` containing Python objects of any type.
+
+For most PyO3 usage the conversion cost is worth paying to get these benefits. As always, if you're not sure it's worth it in your case, benchmark it!
+
+### Returning Rust values to Python
+
+When returning values from `#[pyfunction]`s and `#[pymethods]`, Python-native types (`&PyAny`, `&PyDict` etc.) can be used with zero cost.
```

Comment:
Nit: put the `s` consistently either inside or out (#[pyfunction]s vs #[pymethods])

Reply:
Mmm I agree - sadly that's what the attributes are called! I'll change the wording here.

Reply:
Ah right, my bad.